clang/lib/Driver/OffloadBundler.cpp

```diff
@@ -1040,39 +1040,49 @@ class ArchiveFileHandler final : public FileHandler {
     for (auto &C : Ar->children(Err)) {
       ++ChildIndex;
       auto BinOrErr = C.getAsBinary();
+
+      std::unique_ptr<FileHandler> FH{nullptr};
+      std::unique_ptr<MemoryBuffer> Buf{nullptr};
+
       if (!BinOrErr) {
         if (auto Err = isNotObjectErrorInvalidFileType(BinOrErr.takeError()))
           return Err;
-        continue;
-      }
 
-      auto &Bin = BinOrErr.get();
-      if (!Bin->isObject())
-        continue;
+        /* Handle bundled BC Files */
+        FH = std::make_unique<BinaryFileHandler>(BundlerConfig);
+        auto MR = C.getMemoryBufferRef();
+        assert(MR);
+        Buf = MemoryBuffer::getMemBuffer(*MR, false);
+      } else {
+        auto &Bin = BinOrErr.get();
+        if (!Bin->isObject())
+          continue;
 
-      auto CheckOrErr = CheckIfObjectFileContainsExcludedTargets(C);
-      if (!CheckOrErr)
-        return CheckOrErr.takeError();
+        auto CheckOrErr = CheckIfObjectFileContainsExcludedTargets(C);
+        if (!CheckOrErr)
+          return CheckOrErr.takeError();
 
-      if (*CheckOrErr) {
-        LLVM_DEBUG(outs() << "Add child to ban list. Index: " << ChildIndex
-                          << "\n");
-        ExcludedChildIndexes.emplace(ChildIndex);
-      }
+        if (*CheckOrErr) {
+          LLVM_DEBUG(outs()
+                     << "Add child to ban list. Index: " << ChildIndex << "\n");
+          ExcludedChildIndexes.emplace(ChildIndex);
+        }
 
-      auto Obj = std::unique_ptr<ObjectFile>(cast<ObjectFile>(Bin.release()));
-      auto Buf = MemoryBuffer::getMemBuffer(Obj->getMemoryBufferRef(), false);
+        auto Obj = std::unique_ptr<ObjectFile>(cast<ObjectFile>(Bin.release()));
+        Buf = MemoryBuffer::getMemBuffer(Obj->getMemoryBufferRef(), false);
 
-      // Collect the list of bundles from the object.
-      ObjectFileHandler OFH(std::move(Obj), BundlerConfig);
-      if (Error Err = OFH.ReadHeader(*Buf))
+        // Collect the list of bundles from the object.
```

Comment:
May be we can move this comment to right after if-then-else block?

Reply:
Revised comment and moved it.